common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1061,13 +1061,15 @@ def get_parent_location(self, locator, **kwargs):
             raise ItemNotFoundError(locator)
 
         course = self._lookup_course(locator.course_key)
-        parent_id = self._get_parent_from_structure(BlockKey.from_usage_key(locator), course.structure)
-        if parent_id is None:
+        parent_ids = self._get_parents_from_structure(BlockKey.from_usage_key(locator), course.structure)
+        if len(parent_ids) == 0:
             return None
+        # find alphabetically least
+        parent_ids.sort(key=lambda e: e.type + e.id)
```

Comment:
Nit: you could sort `key=lambda e: (e.type, e.id)`, and thus not have to worry about concatenating strings.